code/datums/gamemode/role/vampire_role.dm

```diff
@@ -222,7 +222,9 @@
 		target.vessel.remove_reagent(BLOOD,blood)
 		var/mob/living/carbon/V = assailant
 		if(V)
-			if(V.nutrition < OVEREAT_THRESHOLD) //Gives the vampire a little bit of food, at a rate of 1/4 the blood sucked.
+			var/fatty_chemicals = target.has_reagent_in_blood(CHEESYGLOOP) || target.has_reagent_in_blood(CORNOIL) //If the target has these chemicals in his blood the vampire can get fat from sucking blood.
```

Comment:
Not really sure about this one, but couldn't you use list(CHEESYGLOOP, CORNOIL) instead of the double checks?

Reply:
Let me see what I can do

Reply:
Fixed